pkg/awsvpc/sg_analysis.go

```diff
@@ -112,9 +110,9 @@ func getRuleStr(direction, connStr, ipRanges string) string {
 // getProtocolICMPRule returns rule results corresponding to the provided rule obj with icmp connection
 func (sga *AWSSGAnalyzer) getProtocolICMPRule(ruleObj *types.IpPermission, direction string) (
 	ruleStr string, ruleRes *commonvpc.SGRule, err error) {
-	minPort := int64(*ruleObj.FromPort)
-	maxPort := int64(*ruleObj.ToPort)
-	conns := commonvpc.GetICMPconn(&minPort, &maxPort)
+	icmpType := int64(*ruleObj.FromPort)
+	icmpCode := int64(*ruleObj.ToPort)
+	conns := connection.ICMPConnection(icmpType, icmpType, icmpCode, icmpCode)
```

Comment:
Is the special case where type or code equal -1 handled?

Reply:
handled